framework/validators/ExistValidator.php

```diff
@@ -110,13 +116,24 @@ public function validateAttribute($model, $attribute)
         $targetClass = $this->targetClass === null ? get_class($model) : $this->targetClass;
         $query = $this->createQuery($targetClass, $conditions);
 
+        $db = $targetClass::getDb();
+        $disabledSlaves = false;
+        if (!$this->canValidateOnSlaveDb && $db->enableSlaves){
+            $db->enableSlaves = false;
+            $disabledSlaves = true;
+        }
```

Comment:
Why not use `Connection::useMaster()`?

Reply:
The code looks messy now

Reply:
@rob006 I did think of using `Connection::useMaster()`, but if you consider the situation when `canValidateOnSlaveDb` is true, you'll find it's not a good choice.

Reply:
> but if you consider the situation when canValidateOnSlaveDb is true, you'll find it's not a good choice.

Why? Obviously you need some conditions here (use `Connection::useMaster()` only when `canValidateOnSlaveDb` is `false`), but you can move it to separate method.

Reply:
Yes, that would make it too, but I my code used same logic within `Connection::useMaster()`, and it's simpler.

Reply:
@rossoneri It does not handle exceptions correctly and at the best case it is just code duplication.